src/langchain_google_alloydb_pg/embeddings.py

```diff
@@ -30,14 +31,43 @@ def __init__(self, engine: AlloyDBEngine, model_id: str):
             engine (AlloyDBEngine): Connection pool engine for managing connections to Postgres database.
             model_id (str): The model id used for generating embeddings.
 
+        Raises:
+            :class:`ValueError`: if model does not exist. Use AlloyDBModelManager to create the model.
+
         """
         self._engine = engine
         self.model_id = model_id
 
-        # TODO: @vishwarajanand - We should validate the model_id here
-        # models = AlloyDBModel(engine, model_id)
-        # if not models.exists():
-        #     raise IllegalArgumentError(f"Model {model_id} does not exist.")
+        self.model_manager = AlloyDBModelManager.create_sync(engine=self._engine)
```

Comment:
This class will be used within Genai applications, therefore I worry about only support a sync init which may slow down the application start time. Can we use the create/create_sync pattern?

Reply:
added